src/gwas.jl

```diff
@@ -379,6 +512,96 @@ function one_fold(
     return myerrors :: Vector{T}
 end
 
+"""
+    one_fold(x::BEDFile, y, path, folds, fold)
+
+If used with a `BEDFile` object `x`, then the additional optional arguments are:
+
+- `pids`, a vector of process IDs. Defaults to `procs(x)`.
+"""
+function one_fold(
+    x        :: SnpLike{2},
+    y        :: Array{T,1},
+    path     :: DenseVector{Int},
+    folds    :: DenseVector{Int},
+    fold     :: Int;
+    pids     :: Vector{Int} = procs(x),
+    tol      :: T    = convert(T, 1e-4),
+    max_iter :: Int  = 100,
+    max_step :: Int  = 50,
+    quiet    :: Bool = true
+) where {T <: Float}
+    # dimensions of problem
+    n,p = size(x)
+    print_with_color(:red, "gwas332, Starting one_fold().\n")
+
+    # make vector of indices for folds
+    test_idx = folds .== fold
+    test_size = sum(test_idx)
+
+    # train_idx is the vector that indexes the TRAINING set
+    train_idx = .!test_idx
+    mask_n    = convert(Vector{Int}, train_idx)
+    mask_test = convert(Vector{Int}, test_idx)
+    #Gordon
+    println("test_size = $(test_size)")
+    #println("train_idx = $(train_idx)")
+    println()
+    println()
+    #println("test_idx = $(test_idx)")
+
+    # compute the regularization path on the training set
+    betas = iht_path(x, y, path, mask_n=mask_n, max_iter=max_iter, quiet=quiet, max_step=max_step, pids=pids, tol=tol)
+
+    # tidy up
+    #gc()
+
+    # preallocate vector for output
+    myerrors = zeros(T, length(path))
+
+    # allocate an index vector for b
+    indices = falses(p)
+
+    # allocate the arrays for the test set
+    xb = SharedArray{T}((n,), init = S -> S[localindexes(S)] = zero(T), pids=pids) :: SharedVector{T}
+    b  = SharedArray{T}((p,), init = S -> S[localindexes(S)] = zero(T), pids=pids) :: SharedVector{T}
+    r  = SharedArray{T}((n,), init = S -> S[localindexes(S)] = zero(T), pids=pids) :: SharedVector{T}
+
+    # compute the mean out-of-sample error for the TEST set
+    # do this for every computed model in regularization path
+    for i = 1:size(betas,2)
+
+        # pull ith model in dense vector format
+        b2 = full(vec(betas[:,i]))
+
+        # copy it into SharedArray b
+        copy!(sdata(b),sdata(b2))
+
+        # indices stores Boolean indexes of nonzeroes in b
+        update_indices!(indices, b)
+
+        # compute estimated response Xb with $(path[i]) nonzeroes
+        #A_mul_B!(xb, x, b, indices, path[i], mask_test, pids=pids)
+
+#        p_tmp = convert(Array{T,1}, path[i]) # Gordon - didn't help
+        #A_mul_B!(xb, x, b, indices, path[i], mask_test)
```

Comment:
`A_mul_B!` already exists for SnpArrays type. You can call it the same was as we called `At_mul_B!` as in `L0_reg`. See documentations here for more info: https://openmendel.github.io/SnpArrays.jl/latest/man/snparray/#linear-algebra-with-snparray

Reply:
Oh wait I think I see what you mean. You need a `A_mul_B!` that accepts 6 arguments. Ok you can leave that to me.